internal/hooks/hook_iterator.go

```diff
@@ -0,0 +1,52 @@
+package hooks
+
+import (
+	"github.com/launchdarkly/go-server-sdk/v7/ldhooks"
+)
+
+type hookIterator struct {
+	reverse    bool
+	cursor     int
+	collection *[]ldhooks.Hook
```

Comment:
Any reason to store a pointer to the slice?

Reply:
I can change this to not be a pointer. I did a few experiments and didn't undo this along the way.